src/charts/people-affected-by-crisis-map.js

```diff
@@ -0,0 +1,154 @@
+import fetchCSVData, { ACTIVE_BRANCH } from '../utils/data';
+
+const MAP_FILE_PATH = `https://raw.githubusercontent.com/devinit/gha-data-visualisations/${ACTIVE_BRANCH}/public/assets/data/world_map.geo.json`;
+const CSV_PATH = `https://raw.githubusercontent.com/devinit/gha-data-visualisations/${ACTIVE_BRANCH}/public/assets/data/map_data_long.csv`;
+
+const matchCountryNames = (csvData, worldData) => {
+  const matchedData = csvData.map((stream) => {
+    const countryObject = worldData.find((feature) => feature.properties.iso_a3 === stream.Country_ID);
+    if (countryObject) {
+      // eslint-disable-next-line no-param-reassign
+      stream.Country_name = countryObject.properties.name;
+    }
+
+    return stream;
+  });
+
+  return matchedData;
+};
+
+const processedData = (countries, processedCountryData) => {
+  const data = [];
+  // eslint-disable-next-line array-callback-return
+  countries.map((country) => {
+    const countryData = {};
+    countryData.name = country;
+    // eslint-disable-next-line array-callback-return
+    processedCountryData.map((stream) => {
+      if (stream.Country_name === country) {
+        countryData[stream.variable] = stream.value;
+      }
+    });
+    data.push(countryData);
+  });
+
+  return data;
+};
+
+const dataInjectedGeoJson = (jsonData, groupedData) =>
+  jsonData.map((feature) => {
+    const matchingCountryData = groupedData.find((countryData) => countryData.name === feature.properties.name);
+    if (matchingCountryData) {
+      // eslint-disable-next-line no-param-reassign
+      feature.properties = {
+        ...feature.properties,
+        ...matchingCountryData,
+      };
+    }
+
+    return feature;
+  });
+
+const getColor = (score) => {
+  switch (score) {
+    case '5':
+      return '#7F1850';
+    case '4':
+      return '#AD1156';
+    case '3':
+      return '#D64279';
+    case '2':
+      return '#E4819B';
+    case '1':
+      return '#F6B9C2';
+    default:
+      return '#E6E1E5';
+  }
+};
+
+const highlightFeature = (e) => {
+  const layer = e.target;
+
+  layer.setStyle({
+    fillColor: 'yellow',
+  });
+
+  if (!window.L.Browser.ie && !window.L.Browser.opera && !window.L.Browser.edge) {
+    layer.bringToFront();
+  }
+  // Bind popup to layer
+  layer.bindPopup(layer.feature.properties.name).openPopup();
+};
+
+const onLegendAdd = () => {
+  const div = window.L.DomUtil.create('div', 'info legend');
+  const legendData = [
+    { score: '5', label: '5 - Very High' },
+    { score: '4', label: '4 - High' },
+    { score: '3', label: '3 - Medium' },
+    { score: '2', label: '2 - Low' },
+    { score: '1', label: '1 - Very Low' },
+  ];
+
+  const legendContent = legendData
+    .map((data) => `<i style="background:${getColor(data.score)}"></i><label>${data.label}</label>`)
+    .join('');
+  div.innerHTML = legendContent;
+
+  return div;
+};
+
+function renderPeopleAffectedByCrisisLeaflet() {
+  const map = window.L.map('map').setView([20, -0.09], 2);
```

Comment:
I'm sure this can take in a node instead of an ID.

Don't use IDs ... we've built our visualisations to be rendered in multiple positions on the same page if needed ... IDs don't support that. So you'll either still need the chart wrapper code you'd removed, or write your own 2 - 3 lines of code for getting the elements by class.

Reply:
The ;leaflet initializer actually only takes in an id. This is where the id is used in the source code...
`initialize: function (id, options) { // (HTMLElement or String, Object)
		options = Util.setOptions(this, options);

		// Make sure to assign internal flags at the beginning,
		// to avoid inconsistent state in some edge cases.
		this._handlers = [];
		this._layers = {};
		this._zoomBoundLayers = {};
		this._sizeChanged = true;

		this._initContainer(id);
		this._initLayout();

		// hack for https://github.com/Leaflet/Leaflet/issues/1980
		this._onResize = Util.bind(this._onResize, this);

		this._initEvents();

		if (options.maxBounds) {
			this.setMaxBounds(options.maxBounds);
		}

		if (options.zoom !== undefined) {
			this._zoom = this._limitZoom(options.zoom);
		}

		if (options.center && options.zoom !== undefined) {
			this.setView(toLatLng(options.center), options.zoom, {reset: true});
		}
`